extension/js/content_scripts/webmail/gmail-element-replacer.ts

```diff
@@ -295,29 +295,34 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     const allContenteditableEls = $("div[contenteditable='true']").not('.evaluated').addClass('evaluated');
     for (const contenteditableEl of allContenteditableEls) {
       const contenteditable = $(contenteditableEl);
-      const draftLinkMatch = contenteditable
+      let draftId = '';
+      const legacyDraftLinkMatch = contenteditable
         .html()
-        .substr(0, 1000)
+        .substring(0, 1000)
         .match(/\[flowcrypt:link:draft_compose:([0-9a-fr\-]+)]/);
-      if (draftLinkMatch) {
-        const [, buttonHrefId] = draftLinkMatch;
-        const button = `<a href="#" class="open_draft_${Xss.escape(buttonHrefId)}">Open draft</a>`;
-        Xss.sanitizeReplace(contenteditable, button);
-        $(`a.open_draft_${buttonHrefId}`).on(
-          'click',
-          Ui.event.handle(target => {
-            if (this.injector.openComposeWin(buttonHrefId)) {
-              closeGmailComposeWindow(target);
-            }
-          })
-        );
+      if (legacyDraftLinkMatch) {
+        const [, legacyDraftId] = legacyDraftLinkMatch;
+        draftId = legacyDraftId;
+      }
+      const draftHtml = contenteditable.html();
+      // Used some hacky way to check if draft is compose draft or draft from thread reply
+      // Reply `table` element (class named aoP) has avatar image(class named ajn). Compose draft doesn't have it
+      // For thread reply draft, we don't need to call openComposeWin
+      // Reply draft window will be replaced by secure compose by replaceStandardReplyBox function
+      const isComposeDraft = $(contenteditable).closest('table.aoP').find('img.ajn').length < 1;
+      if (PgpArmor.isEncryptedMsg(draftHtml) && isComposeDraft) {
+        draftId = String($(contenteditable).closest('.I5').find('input[name="draft"]')?.val())?.split(':')[1] ?? '';
+      }
+      if (draftId) {
         // close original draft window
-        const closeGmailComposeWindow = (target: HTMLElement) => {
+        const closeGmailComposeWindow = (target: HTMLElement | JQuery<HTMLElement>) => {
           const mouseUpEvent = document.createEvent('Event');
           mouseUpEvent.initEvent('mouseup', true, true); // Gmail listens for the mouseup event, not click
-          $(target).closest('.dw').find('.Ha')[0].dispatchEvent(mouseUpEvent); // jquery's trigger('mouseup') doesn't work for some reason
+          $(target).closest('.nH.Hd').find('.Ha')[0].dispatchEvent(mouseUpEvent); // jquery's trigger('mouseup') doesn't work for some reason
```

Comment:
After changing `target` type to `JQuery<HTMLElement>`, this can be changed to `target.closest('.nH.Hd')...`